### PR TITLE
Adc function update

### PR DESCRIPTION
**Pull request by RaulTrombin**
Have a pub enum adc_Channel for user
Old read_adc becomes read_adc_all
read_ad now allow read single channels and differential with a defined set adc_Channel
Update example
New example adc loop

### CODE REVIEW

**Review by patrickelectric — CHANGES_REQUESTED** [review]
Please, break your commits in multiple PRs when possible to avoid having duplicated code between PRs.

**Review by joaoantoniocardoso — CHANGES_REQUESTED** [review]
The code looks good to me, but as #6 is merged, a rebase is needed before merging this PR

File: src/lib.rs [hunk 1]
```diff
@@ -229,16 +226,46 @@ impl Navigator {
...
+                block!(self.adc.read(&mut ads1x1x::channel::DifferentialA0A3)).unwrap()
+            }
+            adc_Channel::DifferentialA1A3 => {
+                block!(self.adc.read(&mut ads1x1x::channel::DifferentialA1A3)).unwrap()
+            }
+            adc_Channel::DifferentialA2A3 => {
+                block!(self.adc.read(&mut ads1x1x::channel::DifferentialA2A3)).unwrap()
+            }
+        }
+    }
```
Comment (patrickelectric) [inline src/lib.rs:267]: why not just making it:
```rs
    pub fn read_adc(&mut self, channel: adc_Channel) -> i16 {
        block!(self.adc.read(&mut channel.into())).unwrap()
    }
```

Comment (joaoantoniocardoso) [reply src/lib.rs:267]: @patrickelectric true, and it seems to be possible to just use the enum itself, as shown [here](https://github.com/eldruin/ads1x1x-rs/blob/master/examples/trait.rs#L12):

```Rust
    pub fn read_adc(&mut self, channel: adc_Channel) -> i16 {
        block!(self.adc.read(channel)).unwrap()
    }
```

Comment (RaulTrombin) [reply src/lib.rs]: Good, changed the imports, using DynamicOneShot the trait cover enum inputs

File: src/lib.rs [hunk 2]
```diff
@@ -225,16 +225,21 @@ impl Navigator {
...
 
-    pub fn read_adc(&mut self) -> ADCData {
+    pub fn read_adc_all(&mut self) -> ADCData {
         ADCData {
             channel: [
-                block!(self.adc.read(&mut channel::SingleA0)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA1)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA2)).unwrap(),
-                block!(self.adc.read(&mut channel::SingleA3)).unwrap(),
+                block!(self.adc.read(adc_Channel::SingleA0)).unwrap(),
```
Comment (patrickelectric) [inline src/lib.rs:231]: why we need to call block in a function that already encapsulates block ?

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 2 above)
Comment (RaulTrombin) [reply src/lib.rs:231]: it works if we remove, 
it's not clear why all the examples from the library use this macro

Comment (RaulTrombin) [reply src/lib.rs:231]: you are right,
linux_embedded_hal

already uses the blocking methods from embedded_hal
https://docs.rs/embedded-hal/latest/embedded_hal/blocking/

Comment (RaulTrombin) [reply src/lib.rs]: did a bit of confusion with block macro vs the embedded_hall::block

**Review by patrickelectric — APPROVED** [review]

**Pull request merged**